Denys Bykov/Task1/StringList.cpp

```diff
@@ -0,0 +1,133 @@
+#include "StringList.hpp"
+
+void StringListInit(StringList* list)
+{
+    *list = static_cast<char**>(malloc(sizeof(char**)));
+    **list = nullptr;
+}
+
+void StringListDestroy(StringList* list)
+{
+    char** pos = *list;
+    for (; *pos; pos++)
+        free(*pos);
+    free(*list);
+    *list = nullptr;
+}
+
+int  StringListSize(StringList list)
+{
+    char** pos = list;
+    for (; *pos; pos++) {}
+    return static_cast<int>(pos - list);
+}
+
+void StringListAdd(StringList* list, char* str)
+{
+    char** memory_block = *list;
+    int old_size = StringListSize(*list);
+    int new_size = old_size + 1;
+    int size_to_allocate = sizeof(char**) * (new_size + 1);
+    memory_block  =
```

Comment:
It is better if whole node data located in separate memory block. Could you please use big O notation and provide complexity of inserting data into you list.

Reply:
Right now my implementation is rather 2d array than a linked list, so everything is allocated in continuous block of memory. As I can see I was wrong and the task is to implement a linked list?

Reply:
The main idea of this task is to learn pointers and pointer arithmetics. It could be an array but in this case better to use Array in name of the function instead of a list. If you named it as list, it should work like a list.